src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -943,7 +943,9 @@ public Solution WithDocumentName(DocumentId documentId, string name)
         /// Creates a new solution instance with the document specified updated to be contained in
         /// the sequence of logical folders.
         /// </summary>
-        public Solution WithDocumentFolders(DocumentId documentId, IEnumerable<string> folders)
+        /// <param name="documentId">Document id.</param>
+        /// <param name="folders">Sequence of folders. <see langword="null"/> values are ignored.</param>
```

Comment:
1. i am glad we're doc'ing this.  however:
2. yikes.  that seems awful.  i'd much rather us throw (and i don't care if that breaks something).

Reply:
I'll remove the filtering and throw when an item is null in a follow-up PR.